src/java/org/apache/cassandra/metrics/TableMetrics.java

```diff
@@ -667,28 +668,38 @@ public Long loadValue()
                                                          () -> combineHistograms(cfs.getSSTables(SSTableSet.CANONICAL), 
                                                                                  SSTableReader::getEstimatedCellPerPartitionCount), null);
 
-        estimatedRowCount = createTableGauge("EstimatedRowCount", "EstimatedRowCount", new Gauge<Long>()
+        estimatedRowCount = createTableGauge("EstimatedRowCount", "EstimatedRowCount", new CachedGauge<>(1, TimeUnit.SECONDS)
         {
-            public Long getValue()
+            public Long loadValue()
             {
                 long memtableRows = 0;
-                for (Memtable memtable : cfs.getTracker().getView().getAllMemtables())
+                OpOrder.Group readGroup = null;
+                try
                 {
-                    if (memtable instanceof TrieMemtable)
+                    for (Memtable memtable : cfs.getTracker().getView().getAllMemtables())
                     {
-                        ColumnFilter.Builder builder = ColumnFilter.allRegularColumnsBuilder(cfs.metadata(), true);
-                        memtableRows += ((TrieMemtable) memtable).rowCount(builder.build(), DataRange.allData(cfs.getPartitioner()));
+                        if (readGroup == null)
+                        {
+                            readGroup = memtable.readOrdering().start();
```

Comment:
I think that it's needed given that `estimateRowCount` will iterate through a sample of rows on the first call - https://github.com/datastax/cassandra/blob/main/src/java/org/apache/cassandra/db/memtable/MemtableAverageRowSize.java#L35-L64.